<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1536</width>
    <height>864</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QListWidget" name="listWidget_lines">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>40</y>
      <width>271</width>
      <height>201</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
      <bold>true</bold>
     </font>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>0</y>
      <width>151</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Цепочки:</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_delete_line">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>190</y>
      <width>241</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Segoe UI</family>
      <pointsize>12</pointsize>
      <italic>false</italic>
      <bold>false</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
    background-color: #4CAF50;  /* Задаем цвет фона */
    font: 12pt &quot;Segoe UI&quot;;
    color: white;  /* Задаем цвет текста */
    border: 1px solid rgb(154, 255, 143);  /* Задаем толщину и цвет границы */
    border-radius: 7px;  /* Закругляем углы */
    box-shadow: 0px 4px 8px rgba(0, 0, 0, 0.1);  /* Добавляем тень */
    transition: background-color 0.3s, color 0.3s, box-shadow 0.3s, transform 0.3s;  /* Добавляем плавные переходы */
}

QPushButton:hover {
    background-color: #45a049;  /* Цвет фона при наведении */
    box-shadow: 0px 8px 16px rgba(0, 0, 0, 0.2);  /* Увеличиваем тень при наведении */
    transform: translateY(-2px);  /* Поднимаем кнопку на 2 пикселя */
}

QPushButton:pressed {
    background-color: #39803d;  /* Цвет фона при нажатии */
    box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.1);  /* Уменьшаем тень при нажатии */
    transform: translateY(2px);  /* Поднимаем кнопку на 2 пикселя при нажатии */
}</string>
    </property>
    <property name="text">
     <string>Удалить цепочку</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_add_lines_ready">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>40</y>
      <width>241</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Segoe UI</family>
      <pointsize>12</pointsize>
      <italic>false</italic>
      <bold>false</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
    background-color: #4CAF50;  /* Задаем цвет фона */
    font: 12pt &quot;Segoe UI&quot;;
    color: white;  /* Задаем цвет текста */
    border: 1px solid rgb(154, 255, 143);  /* Задаем толщину и цвет границы */
    border-radius: 7px;  /* Закругляем углы */
    box-shadow: 0px 4px 8px rgba(0, 0, 0, 0.1);  /* Добавляем тень */
    transition: background-color 0.3s, color 0.3s, box-shadow 0.3s, transform 0.3s;  /* Добавляем плавные переходы */
}

QPushButton:hover {
    background-color: #45a049;  /* Цвет фона при наведении */
    box-shadow: 0px 8px 16px rgba(0, 0, 0, 0.2);  /* Увеличиваем тень при наведении */
    transform: translateY(-2px);  /* Поднимаем кнопку на 2 пикселя */
}

QPushButton:pressed {
    background-color: #39803d;  /* Цвет фона при нажатии */
    box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.1);  /* Уменьшаем тень при нажатии */
    transform: translateY(2px);  /* Поднимаем кнопку на 2 пикселя при нажатии */
}</string>
    </property>
    <property name="text">
     <string>Добавить цепочки</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_stage_1">
    <property name="geometry">
     <rect>
      <x>630</x>
      <y>260</y>
      <width>271</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Segoe UI</family>
      <pointsize>12</pointsize>
      <italic>false</italic>
      <bold>false</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
    background-color: #4CAF50;  /* Задаем цвет фона */
    font: 12pt &quot;Segoe UI&quot;;
    color: white;  /* Задаем цвет текста */
    border: 1px solid rgb(154, 255, 143);  /* Задаем толщину и цвет границы */
    border-radius: 7px;  /* Закругляем углы */
    box-shadow: 0px 4px 8px rgba(0, 0, 0, 0.1);  /* Добавляем тень */
    transition: background-color 0.3s, color 0.3s, box-shadow 0.3s, transform 0.3s;  /* Добавляем плавные переходы */
}

QPushButton:hover {
    background-color: #45a049;  /* Цвет фона при наведении */
    box-shadow: 0px 8px 16px rgba(0, 0, 0, 0.2);  /* Увеличиваем тень при наведении */
    transform: translateY(-2px);  /* Поднимаем кнопку на 2 пикселя */
}

QPushButton:pressed {
    background-color: #39803d;  /* Цвет фона при нажатии */
    box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.1);  /* Уменьшаем тень при нажатии */
    transform: translateY(2px);  /* Поднимаем кнопку на 2 пикселя при нажатии */
}</string>
    </property>
    <property name="text">
     <string>Этап 1</string>
    </property>
   </widget>
   <widget class="QListWidget" name="listWidget_stage_1">
    <property name="geometry">
     <rect>
      <x>630</x>
      <y>70</y>
      <width>271</width>
      <height>171</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
      <bold>true</bold>
     </font>
    </property>
   </widget>
   <widget class="QListWidget" name="listWidget_stage_2">
    <property name="geometry">
     <rect>
      <x>930</x>
      <y>70</y>
      <width>271</width>
      <height>171</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
      <bold>true</bold>
     </font>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_stage_2">
    <property name="geometry">
     <rect>
      <x>930</x>
      <y>260</y>
      <width>271</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Segoe UI</family>
      <pointsize>12</pointsize>
      <italic>false</italic>
      <bold>false</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
    background-color: #4CAF50;  /* Задаем цвет фона */
    font: 12pt &quot;Segoe UI&quot;;
    color: white;  /* Задаем цвет текста */
    border: 1px solid rgb(154, 255, 143);  /* Задаем толщину и цвет границы */
    border-radius: 7px;  /* Закругляем углы */
    box-shadow: 0px 4px 8px rgba(0, 0, 0, 0.1);  /* Добавляем тень */
    transition: background-color 0.3s, color 0.3s, box-shadow 0.3s, transform 0.3s;  /* Добавляем плавные переходы */
}

QPushButton:hover {
    background-color: #45a049;  /* Цвет фона при наведении */
    box-shadow: 0px 8px 16px rgba(0, 0, 0, 0.2);  /* Увеличиваем тень при наведении */
    transform: translateY(-2px);  /* Поднимаем кнопку на 2 пикселя */
}

QPushButton:pressed {
    background-color: #39803d;  /* Цвет фона при нажатии */
    box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.1);  /* Уменьшаем тень при нажатии */
    transform: translateY(2px);  /* Поднимаем кнопку на 2 пикселя при нажатии */
}</string>
    </property>
    <property name="text">
     <string>Этап 2</string>
    </property>
   </widget>
   <widget class="QListWidget" name="listWidget_stage_3">
    <property name="geometry">
     <rect>
      <x>1230</x>
      <y>70</y>
      <width>271</width>
      <height>171</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
      <bold>true</bold>
     </font>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_stage_3">
    <property name="geometry">
     <rect>
      <x>1230</x>
      <y>260</y>
      <width>271</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Segoe UI</family>
      <pointsize>12</pointsize>
      <italic>false</italic>
      <bold>false</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
    background-color: #4CAF50;  /* Задаем цвет фона */
    font: 12pt &quot;Segoe UI&quot;;
    color: white;  /* Задаем цвет текста */
    border: 1px solid rgb(154, 255, 143);  /* Задаем толщину и цвет границы */
    border-radius: 7px;  /* Закругляем углы */
    box-shadow: 0px 4px 8px rgba(0, 0, 0, 0.1);  /* Добавляем тень */
    transition: background-color 0.3s, color 0.3s, box-shadow 0.3s, transform 0.3s;  /* Добавляем плавные переходы */
}

QPushButton:hover {
    background-color: #45a049;  /* Цвет фона при наведении */
    box-shadow: 0px 8px 16px rgba(0, 0, 0, 0.2);  /* Увеличиваем тень при наведении */
    transform: translateY(-2px);  /* Поднимаем кнопку на 2 пикселя */
}

QPushButton:pressed {
    background-color: #39803d;  /* Цвет фона при нажатии */
    box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.1);  /* Уменьшаем тень при нажатии */
    transform: translateY(2px);  /* Поднимаем кнопку на 2 пикселя при нажатии */
}</string>
    </property>
    <property name="text">
     <string>Этап 3</string>
    </property>
   </widget>
   <widget class="QListWidget" name="listWidget_residual_rules">
    <property name="geometry">
     <rect>
      <x>630</x>
      <y>10</y>
      <width>271</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
      <bold>true</bold>
     </font>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_gen">
    <property name="geometry">
     <rect>
      <x>543</x>
      <y>310</y>
      <width>450</width>
      <height>50</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Segoe UI</family>
      <pointsize>12</pointsize>
      <italic>false</italic>
      <bold>false</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
    background-color: #4CAF50;  /* Задаем цвет фона */
    font: 12pt &quot;Segoe UI&quot;;
    color: white;  /* Задаем цвет текста */
    border: 1px solid rgb(154, 255, 143);  /* Задаем толщину и цвет границы */
    border-radius: 7px;  /* Закругляем углы */
    box-shadow: 0px 4px 8px rgba(0, 0, 0, 0.1);  /* Добавляем тень */
    transition: background-color 0.3s, color 0.3s, box-shadow 0.3s, transform 0.3s;  /* Добавляем плавные переходы */
}

QPushButton:hover {
    background-color: #45a049;  /* Цвет фона при наведении */
    box-shadow: 0px 8px 16px rgba(0, 0, 0, 0.2);  /* Увеличиваем тень при наведении */
    transform: translateY(-2px);  /* Поднимаем кнопку на 2 пикселя */
}

QPushButton:pressed {
    background-color: #39803d;  /* Цвет фона при нажатии */
    box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.1);  /* Уменьшаем тень при нажатии */
    transform: translateY(2px);  /* Поднимаем кнопку на 2 пикселя при нажатии */
}</string>
    </property>
    <property name="text">
     <string>Сгенерировать цепочки по грамматике</string>
    </property>
   </widget>
   <widget class="QListWidget" name="listWidget_gen">
    <property name="geometry">
     <rect>
      <x>368</x>
      <y>370</y>
      <width>800</width>
      <height>400</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
      <bold>true</bold>
     </font>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1536</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
